gulpfile.babel.js

```diff
@@ -0,0 +1,43 @@
+import gulp from 'gulp';
+import nodemon from 'gulp-nodemon';
+import mocha from 'gulp-mocha';
+import concat from 'gulp-concat';
+import uglify from 'gulp-uglify';
+import rename from 'gulp-rename';
+import sass from 'gulp-ruby-sass';
+import imagemin from 'gulp-imagemin';
+import cache from 'gulp-cache';
+
+gulp.task('scripts', () => gulp.src(['public/js/**', 'app/**/*.js'])
+  .pipe(concat('main.js'))
+  .pipe(rename({ suffix: '.min' }))
+  .pipe(uglify())
+  .pipe(gulp.dest('build/js')));
+gulp.task('sass', () => sass('public/css/common.scss', { style: 'compressed' })
+  .pipe(rename({ suffix: '.min' }))
+  .pipe(gulp.dest('build/css')));
+gulp.task('images', () => gulp.src('public/img/**/*')
+  .pipe(cache(imagemin({ optimizationLevel: 5, progressive: true, interlaced: true })))
+  .pipe(gulp.dest('build/img')));
+
+gulp.task('server', () => {
+  nodemon({
+    script: 'server.js',
+    watch: ['server.js', 'app.js', 'routes/', 'public/*', 'public/*/**'],
```

Comment:
do we need to restart nodemon if files in public folder changes?
maybe just watch [server.js, _app/**/*_, _config/**/*_] i.e. server files only

we should use a browser reloader to reload browser when client file changes